ql/src/java/org/apache/hadoop/hive/ql/ddl/table/CreateTableOperation.java

```diff
@@ -140,6 +145,26 @@ private void createTableNonReplaceMode(Table tbl) throws HiveException {
       Table createdTable = context.getDb().getTable(tbl.getDbName(), tbl.getTableName());
       DataContainer dc = new DataContainer(createdTable.getTTable());
       context.getQueryState().getLineageState().setLineage(createdTable.getPath(), dc, createdTable.getCols());
+
+      // We did not create the table when moving the data files for a non-partitioned table and
+      // thus could not add a write notification required for a transactional table. Do that
+      // here, after we have created the table. Since this is a newly created table, listing all
+      // the files in the directory and listing only the ones corresponding to the given id
+      // doesn't have much difference.
+      if (!createdTable.isPartitioned() && AcidUtils.isTransactionalTable(createdTable)) {
```

Comment:
why is it not required for partitioned table ?

Reply:
For partitioned table, the logic creates the partitioned table first and then load the data into a partition (not the table). See comments and code around line 7632 in genFileSinkPlan(). So, we do not need to do it for a partitioned table.